lib/datasource/bitbucket-tags/index.ts

```diff
@@ -1,133 +1,133 @@
-}
+  static readonly registryStrategy = 'first';
 
-// getReleases fetches list of tags for the repository
-export async function getReleases({
-  registryUrl,
-  lookupName: repo,
-}: GetReleasesConfig): Promise<ReleaseResult | null> {
-  const cacheKey = getCacheKey(registryUrl, repo, 'tags');
-  const cachedResult = await packageCache.get<ReleaseResult>(
-    cacheNamespace,
-    cacheKey
-  );
-  // istanbul ignore if
-  if (cachedResult) {
-    return cachedResult;
-  }
+  static readonly defaultRegistryUrls = ['https://bitbucket.org'];
 
-  const url = `/2.0/repositories/${repo}/refs/tags`;
-
-  const bitbucketTags = (
-    await bitbucketHttp.getJson<utils.PagedResult<BitbucketTag>>(url)
-  ).body;
-
-  const dependency: ReleaseResult = {
-    sourceUrl: `${ensureTrailingSlash(getRegistryURL(registryUrl))}${repo}`,
-    releases: null,
-  };
-  dependency.releases = bitbucketTags.values.map(({ name, target }) => ({
-    version: name,
-    gitRef: name,
-    releaseTimestamp: target?.date,
-  }));
-
-  const cacheMinutes = 10;
-  await packageCache.set(cacheNamespace, cacheKey, dependency, cacheMinutes);
-  return dependency;
-}
+  static readonly cacheNamespace = 'datasource-bitbucket';
 
-// getTagCommit fetched the commit has for specified tag
-async function getTagCommit(
-  registryUrl: string,
-  repo: string,
-  tag: string
-): Promise<string | null> {
-  const cacheKey = getCacheKey(registryUrl, repo, `tag-${tag}`);
-  const cachedResult = await packageCache.get<string>(cacheNamespace, cacheKey);
-  // istanbul ignore if
-  if (cachedResult) {
-    return cachedResult;
+  constructor() {
+    super(BitBucketTagsDatasource.id);
   }
 
-  const url = `/2.0/repositories/${repo}/refs/tags/${tag}`;
-
-  const bitbucketTag = (await bitbucketHttp.getJson<BitbucketTag>(url)).body;
-
-  const hash = bitbucketTag.target.hash;
-
-  const cacheMinutes = 10;
-  await packageCache.set(cacheNamespace, cacheKey, hash, cacheMinutes);
+  static getRegistryURL(registryUrl: string): string {
+    // fallback to default API endpoint if custom not provided
+    return registryUrl ?? this.defaultRegistryUrls[0];
+  }
 
-  return hash;
-}
+  static getCacheKey(registryUrl: string, repo: string, type: string): string {
+    return `${BitBucketTagsDatasource.getRegistryURL(
+      registryUrl
+    )}:${repo}:${type}`;
+  }
 
-// getDigest fetched the latest commit for repository main branch
-// however, if newValue is provided, then getTagCommit is called
-export async function getDigest(
-  { lookupName: repo, registryUrl }: DigestConfig,
-  newValue?: string
-): Promise<string | null> {
-  if (newValue?.length) {
-    return getTagCommit(registryUrl, repo, newValue);
+  // getReleases fetches list of tags for the repository
+  @cache({
+    namespace: BitBucketTagsDatasource.cacheNamespace,
+    key: ({ registryUrl, lookupName }: GetReleasesConfig) =>
+      BitBucketTagsDatasource.getCacheKey(registryUrl, lookupName, 'tags'),
+    ttlMinutes: 10,
```

Comment:
What do you think about extending this to the default of 30 minutes?

Reply:
Same for all the TTLs below as well.

Reply:
I see no reason not to. At the moment they are simply mimicking the previous logic.